tgui/packages/tgui/interfaces/PlayerPreferences.js

```diff
@@ -0,0 +1,1092 @@
+                maxLength={1024}
+                value={exploitsDesc}
+                onChange={(e, value) => setExploitsDesc(value)}
+              />
+            </Section>
+          </Grid.Column>
+        </Grid>
+      </Section>
+    </>
+  );
+};
+
+const ProfilePicture = (props, context) => {
+  const { act, data, config } = useBackend(context);
+  return (
+    <ByondUi
+      style={{ width: '400px', height: '100px' }}
+      params={{
+        id: 'player_pref_map',
+        parent: config.window,
+        type: 'map',
+        zoom: 3,
+      }}
+    />
+  );
+};
+
+const TextFieldPreference = (props, context) => {
+  const { act, data, config } = useBackend(context);
+  const { label, value, action, extra, onFocus, noAction } = props;
+  const itemLabel = label || value;
+
+  const handler = noAction ? () => {} : act;
+
+  return (
+    <LabeledList.Item label={itemLabel}>
+      <Input
+        placeholder={data[value] || ''}
+        value={data[value] || value}
+        onFocus={() => onFocus && onFocus()}
+        onChange={(e, newValue) =>
+          !onFocus && handler(action || value, { newValue })}
+      />
+      {extra ? extra : null}
+    </LabeledList.Item>
+  );
+};
+const SelectFieldPreference = (props, context) => {
+  const { act, data, config } = useBackend(context);
+  const { label, value, options, action } = props;
+  const itemLabel = label || value;
+  let parsedOptions = Object.values(data[options] || options);
+  if (Array.isArray(options)) {
+    parsedOptions = options;
+  } else if (typeof data[options] === 'object') {
+    parsedOptions = Object.values(data[options]);
+  }
+
+  return (
+    <LabeledList.Item label={itemLabel}>
+      <Dropdown
+        width="95%"
+        options={parsedOptions}
+        selected={data[value]}
+        onSelected={newValue => act(action || value, { newValue })}
+      />
+    </LabeledList.Item>
+  );
+};
+
+const ToggleFieldPreference = (props, context) => {
+  const { act, data, config } = useBackend(context);
+  const {
+    label,
+    value,
+    leftValue = 1,
+    leftLabel = 'True',
+    rightValue = 0,
+    rightLabel = 'False',
+    action,
+  } = props;
+  const itemLabel = label || value;
+
+  let labelLeft = leftLabel || leftValue;
+  let labelRight = rightLabel || rightValue;
+
+  return (
+    <LabeledList.Item label={itemLabel}>
+      <Button.Checkbox
+        inline
+        content={labelLeft}
+        checked={data[value] === leftValue}
+        onClick={() => act(action)}
+      />
+      <Button.Checkbox
+        inline
+        content={labelRight}
+        checked={data[value] === rightValue}
+        onClick={() => act(action)}
+      />
+    </LabeledList.Item>
+  );
+};
+
+const JobPreferences = (props, context) => {
+  const { act, data, config } = useBackend(context);
+  const [shownDescription, setShownDescription] = useLocalState(
+    context,
+    'shown-desc',
+    null
+  );
+
+  const xenoJobs = ['Xeno Queen', 'Xenomorph'];
```

Comment:
```suggestion
  const xenoJobs = ['Hive Leader', 'Xenomorph'];
```
Xenomorph Queen is a bit of a misnomer, it can be either a Shrike or a Queen.

Reply:
Its just what it was called internally, but i can update that too

Reply:
Changing this will have the side effect of unbanning people from xeno queens